src/utils/bridgeMonitor.js

```diff
@@ -0,0 +1,183 @@
+const config = require('config');
+const axios = require('axios');
+const logger = require('winston');
+const { DonationStatus, DONATION_BRIDGE_STATUS } = require('../models/donations.model');
+const { MilestoneStatus } = require('../models/milestones.model');
+const { getTransaction } = require('../blockchain/lib/web3Helpers');
+
+const bridgeMonitorBaseUrl = config.get('bridgeMonitorBaseUrl');
+const getDonationStatusFromBridge = async ({ txHash, tokenAddress }) => {
+  /**
+   * If you want to see an example of bridge monitor response you can check this link
+   * @see{@link https://feathers.bridge.beta.giveth.io/payments?event.returnValues.reference=0xe5f676f6d24fbe43feddb80dcebbedf86e35835b1c5dd984cda46faff73dd98a}
+   * @type {*}
+   */
+  const result = await axios.get(`${bridgeMonitorBaseUrl}/payments`, {
+    params: {
+      'event.returnValues.reference': txHash,
+    },
+  });
+  if (result.data && result.data.data && result.data.data.length >= 0) {
+    const matchedEvent = result.data.data.find(bridgeEvent => {
+      const normalizedToken =
+        tokenAddress === '0x0'
+          ? '0x0000000000000000000000000000000000000000'
+          : tokenAddress.toLowerCase();
+      return bridgeEvent.event.returnValues.token.toLowerCase() === normalizedToken;
+    });
+    return matchedEvent;
+  }
+  return undefined;
+};
+
+const updateMilestoneStatusToPaid = async (app, donation) => {
+  const donationService = app.service('donations');
+  const milestoneService = app.service('milestones');
+  const milestone = await milestoneService.get(donation.ownerTypeId);
+  if (milestone.status !== MilestoneStatus.PAID) return;
+  const donations = await donationService.find({
+    paginate: false,
+    query: {
+      ownerTypeId: donation.ownerTypeId,
+      status: { $in: [DonationStatus.COMMITTED, DonationStatus.PAYING, DonationStatus.PAID] },
```

Comment:
It's not necessary, if milestone status is PAID not COMMITTED or PAYING donation exists.

Reply:
done.